src/Core/Domain/Shipment/QueryResult/OrderShipment.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Shipment\QueryResult;
+
+use DateTime;
+
+class OrderShipment
+{
+    /**
+     * @var int
+     */
+    private int $orderId;
+
+    /**
+     * @var int
+     */
+    private int $carrierId;
+
+    /**
+     * @var int
+     */
+    private int $addressId;
+
+    /**
+     * @var float
+     */
+    private float $shippingCostTaxExcluded;
+
+    /**
+     * @var float
+     */
+    private float $shippingCostTaxIncluded;
+
+    /**
+     * @var string
+     */
+    private ?string $trackingNumber;
+
+    /**
+     * @var DateTime
+     */
+    private ?DateTime $shippedAt;
+
+    /**
+     * @var DateTime
+     */
+    private ?DateTime $deliveredAt;
+
+    /**
+     * @var DateTime
+     */
+    private ?DateTime $cancelledAt;
+
+    /**
+     * @var OrderShipmentProduct[]
+     */
+    private array $products;
+
+    public function __construct(
+        int $orderId,
+        int $carrierId,
+        int $addressId,
+        float $shippingCostTaxExcluded,
+        float $shippingCostTaxIncluded,
```

Comment:
The native float in PHP have precision issues, that is why we introduced the `DecimalNumber` object which is more accurate

One of our good practice for all the CQRS commands and queries is to never use float but `DecimalNumber` instead, that's one of the rare exceptions when we use a non-native type in CQRS

So can you replace this float type with `DecimalNumber` please?